test/qtismtest/runtime/storage/serializable/QtSerializableWithLocalFilesystemDriver.php

```diff
@@ -0,0 +1,428 @@
+use qtism\common\datatypes\QtiPair;
+use qtism\common\datatypes\QtiPoint;
+use qtism\common\enums\BaseType;
+use qtism\common\enums\Cardinality;
+use qtism\data\QtiComponentIterator;
+use qtism\data\storage\xml\XmlCompactDocument;
+use qtism\runtime\common\MultipleContainer;
+use qtism\runtime\common\ResponseVariable;
+use qtism\runtime\common\State;
+use qtism\runtime\storage\binary\LocalQtiBinaryStorage;
+use qtism\runtime\storage\driver\FilesystemLocalDriver;
+use qtism\runtime\storage\serializable\PhpSerializer;
+use qtism\runtime\storage\serializable\QtiSerializableStorage;
+use qtism\runtime\tests\AssessmentItemSessionState;
+use qtism\runtime\tests\AssessmentTestSession;
+use qtism\runtime\tests\AssessmentTestSessionState;
+use qtism\runtime\tests\SessionManager;
+use qtismtest\QtiSmTestCase;
+
+class QtSerializableWithLocalFilesystemDriver extends QtiSmTestCase
+{
+    public function testLocalQtiBinaryStorage(): void
+    {
+        $doc = new XmlCompactDocument();
+        $doc->load(self::samplesDir() . 'custom/runtime/itemsubset.xml');
+        $test = $doc->getDocumentComponent();
+
+        $sessionManager = new SessionManager(new FileSystemFileManager());
+        $serializer = new PhpSerializer();
+        $localStorageDriver = new FilesystemLocalDriver();
+        $storage = new QtiSerializableStorage($sessionManager, $test, $serializer, $localStorageDriver);
+        $session = $storage->instantiate();
+        $sessionId = $session->getSessionId();
+
+        // Instantiating the test session does not mean it is persisted. At the moment,
+        // it is not persistent yet.
+        $this::assertFalse($storage->exists($sessionId));
+
+        $this::assertInstanceOf(AssessmentTestSession::class, $session);
+        $this::assertEquals(AssessmentTestSessionState::INITIAL, $session->getState());
+
+        // The candidate begins the test session at 13:00:00.
+        $session->setTime(new DateTime('2014-07-14T13:00:00+00:00', new DateTimeZone('UTC')));
+        $session->beginTestSession();
+
+        // A little bit of noisy persistence...
+        $storage->persist($session);
+
+        // Now the test is persisted, we can try to know whether it exists in storage.
+        $this::assertTrue($storage->exists($sessionId));
+
+        // Let's retrive the session from storage.
+        $session = $storage->retrieve($sessionId);
+
+        $this::assertEquals(AssessmentTestSessionState::INTERACTING, $session->getState());
+
+        // The test session has begun. We are in linear mode so that all
+        // item sessions must be initialized and selected for presentation
+        // to the candidate.
+        $itemSessionStore = $session->getAssessmentItemSessionStore();
+        $itemSessionCount = 0;
+        $iterator = new QtiComponentIterator($doc->getDocumentComponent(), ['assessmentItemRef']);
+        foreach ($iterator as $itemRef) {
+            $refItemSessions = $itemSessionStore->getAssessmentItemSessions($itemRef);
```

Comment:
it's failed there because inside we have SplObjectStorage and can't find providedItemRef.
it's require refresh mapped item ref links like there https://github.com/oat-sa/qti-sdk/blob/f22687829e58f8538fb219ebb1a64e9a17a9a7cf/src/qtism/runtime/storage/binary/AbstractQtiBinaryStorage.php#L309